app/script/cryptography/CryptographyRepository.coffee

```diff
@@ -429,37 +248,35 @@ class z.cryptography.CryptographyRepository
 
   ###
   Encrypt the generic message for a given session.
-  @note We created the convention that whenever we fail to encrypt for a specific client, we send a bomb emoji (no fun!)
+  @note We created the convention that whenever we fail to encrypt for a specific client, we send a Bomb Emoji (no fun!)
 
   @private
-  @param cryptobox_session [cryptobox.CryptoboxSession] Cryptographic session
   @param generic_message [z.proto.GenericMessage] ProtoBuffer message
   @return [String] Encrypted message as BASE64 encoded string
   ###
-  _encrypt_payload_for_session: (cryptobox_session, generic_message) ->
-    try
-      generic_message_encrypted = cryptobox_session.encrypt generic_message.toArrayBuffer()
-      generic_message_encrypted_base64 = z.util.array_to_base64 generic_message_encrypted
-      @save_session cryptobox_session
-      return generic_message_encrypted_base64
-    catch error
-      ids = z.client.Client.dismantle_user_client_id cryptobox_session.id
-      # Note: We created the convention that whenever we fail to encrypt for a specific client, we send a bomb emoji (no fun!)
-      @logger.error "Failed encrypting '#{generic_message.content}' message for client '#{ids.client_id}' of user '#{ids.user_id}': #{error.message}", error
-      return '💣'
-
-
-  ###############################################################################
-  # Decryption
-  ###############################################################################
+  _encrypt_payload_for_session: (session_id, generic_message) ->
+    return Promise.resolve().then =>
+      values = z.client.Client.dismantle_user_client_id session_id
+      @cryptobox.encrypt session_id, generic_message.toArrayBuffer()
+      .then (generic_message_encrypted) ->
+        values.encrypted = z.util.array_to_base64 generic_message_encrypted
+        return values
```

Comment:
docs say values is a string. why not returning the result or undefined and do we need to use the dismantle_user_client_id here. all we need to encrypt is the session_id and the plaintext